src/vm/ir2jacscript.ts

```diff
@@ -0,0 +1,218 @@
+import jsep from "jsep"
+import { serviceSpecificationFromClassIdentifier } from "../jdom/spec"
+import { VMBase, VMCommand, VMHandler, VMIfThenElse, VMProgram } from "./ir"
+
+// TODO:
+// - insert read() after register in a read context
+// - write a register with fields
+
+export type JacScriptProgram = {
+    program: string[]
+    debug: string[]
+}
+
+function sanitize(s: string) {
+    return s.replace(" ", "_")
+}
+
+function processExpression(e: jsep.Expression, asRead = true): [string, string[]] {
```

Comment:
in general it's better to use object with field names, as it is self documenting the meaning of the values. You can use destructuring on the consumer side
```
.... : { expr: string, vars: string[] }
```

and 
```
const { expr, vars } = ...
```